### ISSUE TITLE
Move selection logic to a plugin

### ISSUE THREAD

**Issue by LaurentGoderre** [vakata/jstree#2041]
By default the plugin adds `aria-selected` to all nodes, which is tricky when it comes to the checkbox node. To simplify this, the selection logic should be move to a plugin to allow better control of behavior in plugins. Since the plugin has an exhaustive set of events, this should be somewhat easy.

**Comment by LaurentGoderre** [comment]
BTW I am working on a PR for this.

**Comment by vakata** [comment]
If you want to detach checkboxes from selection - there is the ```checkboxes.tie_selection``` config option. If it is turned on selected and checked are two separate statuses, if not - they are one and the same, and as such - if you select the root node all nodes will become selected (hence the aria-selected attribute). If this is not desired - detach selected and checked by using the config option.

Anyway - aria-selected is not applied to all nodes - it is only applied to selected nodes.

If I misunderstood something - please explain and I will reopen.

**Comment by LaurentGoderre** [comment]
The issue in question is multi-faceted and somewhat complex so here's my first crack at explaining it.

The core of the issue is that when the checkbox plugin is used, `aria-checked` should be used, not `aria-selected` for several reasons.

1) `aria-checked` is the only way to properly represent three states checkboxes with its values of `true` `false` and `mixed` (What jsTree calls undetermined)

2) A user with low vision that uses a screen reader who perceive the checkbox will expect it to behave and be announced as a checkbox. If it isn't it may cause considerable confusion.

As mentioned in the documentation (for the `tie_selection` setting) mixing selection and check is not a good idea and can lead to a lot opf confusion.

The core plugin adds `aria-selected` to all node which is an issue when trying to implement `aria-checked`. By separating the selection logic into a plugin you get the following benefits.

1) You allow having a tree with no selection (for display purpose only)
2) You allow better control of aria states for the checkbox plugin.

I understand that this is a breaking change but I believe it is required to improve the accessibility of this component.

P.S. I am truly impressed by the amount of work that went into this plugin and I only aim at improving its functionality.
